### ISSUE TITLE
Issue with excludePaths not being consistent with breadcrumbs: null

### ISSUE THREAD

**Issue by darewreck54** [icd2k3/react-router-breadcrumbs-hoc#77]
I think there is an issue with the excludePaths.  

```
const routesConfig: BreadcrumbsRoute[] = [
  {
    path: '/path2',
    breadcrumb: null,
  },
  {
    path: '/path2/:organization',
    breadcrumb: null
  },
  {
    path: '/path2/:organization/:userId',
    breadcrumb: () => 'organization userId'
  }
];

```
In the above example, '/path2/:organization' & '/path2' will not render which is what i want and '/path2/:organization/:userId' will render.

I expected to get the same results if i went with excludePaths:

```
const excludePaths = [
  '/path2/:organization',
  '/path2',
];
```

Based on the readme, I expected the two to reproduce the same result, but that is not the case.

Thanks,
Derek

**Comment by darewreck54** [comment]
Also is there a way to exclude everything and have a includePaths, similar to excludePaths but the opposite?

**Comment by icd2k3** [comment]
> Based on the readme, I expected the two to reproduce the same result, but that is not the case.

This does seem odd... I'll check into this soon.

> Also is there a way to exclude everything and have a includePaths, similar to excludePaths but the opposite?

For this use case, I'd recommend using the `disableDefaults: true` option and then just adding `breadcrumb` to the paths you want to include

**Comment by darewreck54** [comment]
> For this use case, I'd recommend using the `disableDefaults: true` option and then just adding `breadcrumb` to the paths you want to include

What do you mean by "adding breadcrumb" to your path?  If you disable defaults, even if i pass in breadcrumbs, it gets ignored.  Maybe I'm doing something wrong?

**Comment by icd2k3** [comment]
> What do you mean by "adding breadcrumb" to your path? If you disable defaults, even if i pass in breadcrumbs, it gets ignored. Maybe I'm doing something wrong?

```js
withBreadcrumbs(
  [
    {
      path: '/example',
      breadcrumb: () => 'Example'
    },
    {
      path: '/nobreadcrumb',
    }
  ],
  { disableDefaults: true }
)
```

In this case, the only breadcrumb that would render would be `/example`.

**Comment by icd2k3** [comment]
Back to the `excludePaths` option... there does seem to be a bug somewhere here with deep routes w/ params. Here's a demo:
https://codesandbox.io/s/react-router-breadcrumbs-hoc-demo-wgzmj

In this case, I would expect the "Org Name" breadcrumb _not_ to render, but it does. I'll have to spend some time investigating this.

**Comment by darewreck54** [comment]
> ```js
> withBreadcrumbs(
>   [
>     {
>       path: '/example',
>       breadcrumb: () => 'Example'
>     },
>     {
>       path: '/nobreadcrumb',
>     }
>   ],
>   { disableDefaults: true }
> )
> ```

If it's all disabled, why does it need to be included?

```
withBreadcrumbs(
  [
    {
      path: '/example',
      breadcrumb: () => 'Example'
    }
  ],
  { disableDefaults: true }
)
```

shouldn't this work?

An example of this is say you have the path : 'a/b/c/d', but in our your routes you only map 'a/b' and not 'a'.  That means, I would have to define

path: '/a'
path: '/a/b'
path: '/a/b/c'

It would be nice to only have to specify the paths that are actually mapped with 

path: '/a/b'
disableDefaults: true

Does that make sense?

**Comment by darewreck54** [comment]
Also you can't leave breadcrumb undefined based on the type definition

```
export interface BreadcrumbsRoute {
  path: string;
  breadcrumb: React.ReactNode | string;
  matchOptions?: MatchOptions;
}
```

So you have to call out the null manually as well which kind of defeats the purpose to the disableDefaults?

**Comment by icd2k3** [comment]
The issue with `excludePaths` has been fixed as of `3.2.3` (it was an issue with url parameters used in conjunction with `excludePaths`). Here is an updated example of that same codesandbox above, but with the new version applied: https://codesandbox.io/s/react-router-breadcrumbs-hoc-demo-tovug
